newIDE/app/src/Leaderboard/useLeaderboardReplacer.js

```diff
@@ -0,0 +1,415 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import * as React from 'react';
+import Dialog, { DialogPrimaryButton } from '../UI/Dialog';
+import FlatButton from '../UI/FlatButton';
+import { Line } from '../UI/Grid';
+import Text from '../UI/Text';
+import LinearProgress from '@material-ui/core/LinearProgress';
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  TableHeaderColumn,
+  TableRow,
+  TableRowColumn,
+} from '../UI/Table';
+import { ColumnStackLayout } from '../UI/Layout';
+import AuthenticatedUserContext, {
+  type AuthenticatedUser,
+} from '../Profile/AuthenticatedUserContext';
+import { duplicateLeaderboard } from '../Utils/GDevelopServices/Play';
+import { registerGame } from '../Utils/GDevelopServices/Game';
+import { toNewGdMapStringString } from '../Utils/MapStringString';
+
+const gd = global.gd;
+
+type ReplacePromptDialogProps = {|
+  leaderboardsToReplace: ?Array<string>,
+  onClose: () => void,
+  onTriggerReplace: () => void,
+|};
+
+export const ReplacePromptDialog = ({
+  leaderboardsToReplace,
+  onClose,
+  onTriggerReplace,
+}: ReplacePromptDialogProps) => {
+  const { authenticated, onLogin } = React.useContext(AuthenticatedUserContext);
+
+  return (
+    <Dialog
+      actions={
+        authenticated
+          ? [
+              <FlatButton
+                primary
+                key="register-game-later"
+                label={<Trans>Register later</Trans>}
+                onClick={onClose}
+              />,
+              <DialogPrimaryButton
+                primary
+                key="register-game-now"
+                label={<Trans>Register now</Trans>}
+                onClick={onTriggerReplace}
+              />,
+            ]
+          : [
+              <FlatButton
+                primary
+                key="login-later"
+                label={<Trans>Login later</Trans>}
+                onClick={onClose}
+              />,
+              <DialogPrimaryButton
```

Comment:
Should there be a "Create account" action as well?

Reply:
The dialog will allow to create an account, but we should maybe even redirect to create an account first (unlikely a user has an account but is not connected, compared to the 99% of people who will not have an account).